pkg/controller/machineset/controller.go

```diff
@@ -109,13 +109,7 @@ type ReconcileMachineSet struct {
 
 func (r *ReconcileMachineSet) MachineToMachineSets(o handler.MapObject) []reconcile.Request {
 	result := []reconcile.Request{}
-	m := &clusterv1alpha1.Machine{}
-	key := client.ObjectKey{Namespace: o.Meta.GetNamespace(), Name: o.Meta.GetName()}
-	err := r.Client.Get(context.Background(), key, m)
-	if err != nil {
-		klog.Errorf("Unable to retrieve Machine %v from store: %v", key, err)
-		return nil
-	}
+	m, _ := o.Object.(*clusterv1alpha1.Machine)
```

Comment:
If you're not checking if the type assertion succeeded, you can replace `m, _` with `m`

Reply:
We should definitely check the assertion to avoid any runtime panics that could affect other controllers as well

Reply:
@jichenjc hi! do you think you can update this with a type assertion?

Reply:
I missed this ,sorry, I will do update accordingly, thanks for the reminder